sljit_src/sljitConfigInternal.h

```diff
@@ -324,18 +324,19 @@ extern "C" {
 /****************************/
 
 #if (!defined SLJIT_CACHE_FLUSH && defined __has_builtin)
-#if __has_builtin(__builtin___clear_cache)
+#if __has_builtin(__builtin___clear_cache) && !defined(__clang__)
```

Comment:
I don't get the `!defined(__clang__)` check here. Is that function always wrong on clang?

Reply:
gcc >= 10 will have both, but not clang.

clang's implementation of the builtin works fine (tested also in sparc, through NetBSD)

Reply:
Ok, I misunderstood the check. The "will" means it has not been available yet? I mean the comment says `gcc >= 10` but  I don't see `__GNUC__ ` check.

Reply:
__GNUC__ is also defined by clang and icc (at least the latest version that is based on clang and probably also in the older version), hence why checking for __has_builtin is a better indication.

you are correct though, that there might be  compiler that is not based in clang (which has a working builtin) and that also supports __has_builtin without being gcc; but I am willing to take that risk and bet their builtin also works (unlike gcc's)